civis/civis.py

```diff
@@ -359,9 +359,11 @@ class APIClient(MetaMixin):
         - ``'pandas'`` Returns a :class:`pandas:pandas.DataFrame` for
           list-like responses and a :class:`pandas:pandas.Series` for single a
           json response.
-    retry_total : int, optional
+    retry_total : DEPRECATED int, optional
         A number indicating the maximum number of retries for 429, 502, 503, or
-        504 errors.
+        504 errors. This parameter no longer has any effect since v1.15.0,
+        as retries are automatically handled. This parameter will be removed
+        at version 2.0.0.
```

Comment:
Q: what mechanism, if any, is there to ensure we remember to remove it with version 2.0.0?

Reply:
No mechanism other than the maintainers remembering to audit the codebase and handle it plus other API-breaking changes; the `deprecate_param` function sort-of helps us find these things more easily. There's no timeline for a v2.0.0 release yet.